File: Crio/crio_programming_interview_problems-master/LongestCommonPrefix/LongestCommonPrefix.py

# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
def prefix(word1 ,word2):
    i, j = 0, 0 
    while i < len(word1) and j < len(word2):
        if word2[j] == word1[i]:
            i+=1
            j+=1
        else:
            break
    return word1[:i]
# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

def longestCommonPrefix(strs):
    # CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    if len(strs) == 0:
        return ""
    word = strs[0]
    for word2 in strs[1:]:
        word = prefix(word, word2)
    return word
    # CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

if __name__ == '__main__':
    strs = input().split()
    result = longestCommonPrefix(strs)
    print(result)

File: Crio/crio_programming_interview_problems-master/LongestCommonPrefix/LongestCommonPrefix.cpp
#include <bits/stdc++.h>

using namespace std;

class LongestCommonPrefix {
public:
    string longestCommonPrefix(vector<string>& strs) {
        // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        int i,j,mini=INT_MAX;
        for(i=0;i<int(strs.size());i++){
            int n=strs[i].size();
            mini=min(mini,n);}
        string ans="";
        if(mini==INT_MAX)
            mini=0;
        for(j=0;j<mini;j++)
        {
            bool flag=true;
            char d=strs[0][j];
            for(i=1;i<int(strs.size());i++)
                if(strs[i][j]!=d)
                {
                    flag=false;
                    break;
                }
            if(flag)
                ans+=d;
            else
                break;
        }
        return ans;
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
    }
};

int main() {
	string s, word;

	getline(cin, s);
	vector<string> strs;
	stringstream Str(s);
	while(getline(Str, word, ' ')) {
		strs.push_back(word);
	}

	string result = LongestCommonPrefix().longestCommonPrefix(strs);
	cout << result;
	return 0;
}
